File: zhaoxin_ws_fangzhen/src/practice/src/property_talker.cpp
#include <ros/ros.h>
#include <practice/my_property.h>

int main(int argc, char **argv) {
    ros::init(argc, argv, "property_talker");
    ros::NodeHandle nh;
    ros::Publisher pub = nh.advertise<practice::my_property>("property", 10);
    ros::Rate rate(1);  // 1 Hz

    ros::Time start_time = ros::Time::now();

    while (ros::ok()) {
        practice::my_property msg;
        msg.name = "Student";
        msg.Class = 2023;
        ros::Time current_time = ros::Time::now();
        pub.publish(msg);
        ROS_INFO("Published: name = %s, class = %d, at %d", msg.name.c_str(), msg.Class, (current_time - start_time).toSec());
        rate.sleep();
    }

    return 0;
}

File: zhaoxin_ws_fangzhen/src/practice/scripts/property_talker.py
#!/usr/bin/env python3

import rospy
import time
from practice.msg import my_property

def property_talker():
    rospy.init_node('property_talker')
    pub = rospy.Publisher('property', my_property, queue_size=10)
    rate = rospy.Rate(1)  # 1 Hz
    py_start_time = rospy.Time.from_sec(time.time())

    while not rospy.is_shutdown():
        msg = my_property()
        msg.name = "Student"
        msg.Class = 2023
        py_time = rospy.Time.from_sec(time.time())
        pub.publish(msg)
        rospy.loginfo("Published: name = %s, class = %d, at %d", msg.name, msg.Class, py_time.secs - py_start_time.secs)
        rate.sleep()

if __name__ == '__main__':
    try:
        property_talker()
    except rospy.ROSInterruptException:
        pass